### PR TITLE
[Azure Communication Services][CallAutomation]Create Customcontext class

### PR DESCRIPTION
**Pull request by fangchen0601**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

#customers' feedback:
1.	SIP UUI format: The format used by ACS (userToUser) is not aligned to  RFC format.  Developers being more familiar with the latter, I recommend we align to RFC format (User-to-User) in SDK as well as EG payload, just we like we do for custom headers.   

2.	In line with No.1, it might be a better experience to manage the strict header key format at SDK layer, so that every developer doesnt have to remember to set the correct format in the request. 
eg. Instead of callInvite.SipHeaders.Add("userToUser", "616d617a6f6e5f6368696;encoding=hex");
we do callInvite.SipHeaders.AddUserToUser("616d617a6f6e5f6368696;encoding=hex");

eg. Instead of callInvite.SipHeaders.Add("X-MS-Custom-<suffix>", "434erdr454tf"),
we do callInvite.SipHeaders.AddCustomHeader("<suffix>", "434erdr454tf");

#Design:
1.1.1	PSTN target
```
CallInvite callInvite = new CallInvite(new PhoneNumberIdentifier(targetParticipant), callerIdNumber);
callInvite.CustomContext.Add(new SIPUUIHeader(“jason”));
callInvite.CustomContext.Add(new SIPCustomHeader("Frank", "Rizzo")); // SDK will add prefix "X-MS-Custom-"
```
Then SDK will send this payload to PMA:
```
customContext: {        
        "sipHeaders":
        {
                "User-to-User": "jason",
                "X-MS-Custom-Frank": "Rizzo"
        }
}
```

1.1.2	VoiP Target
```
CallInvite callInvite = new CallInvite(new CommunicationUserIdentifier(targetParticipant))
callInvite.CustomContext.Add(new VoipHeader("Foo-1", "Bar")); //no prefix
```
	
Then SDK will send below payload to PMA:
```
customContext: {        
        "voipHeaders":
        {
                "Foo-1": "Bar"
        }
}
```

### CODE REVIEW

**Review by juntuchen-msft — APPROVED** [review]

File: sdk/communication/Azure.Communication.CallAutomation/src/Models/CustomContext.cs [hunk 1]
```diff
@@ -0,0 +1,65 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+
+namespace Azure.Communication.CallAutomation.Models
```
Comment (minwoolee-msft) [inline sdk/communication/Azure.Communication.CallAutomation/src/Models/CustomContext.cs:7]: Need to drop .Models in namespace

Comment (minwoolee-msft) [reply sdk/communication/Azure.Communication.CallAutomation/src/Models/CustomContext.cs:7]: Applies to all other new models

File: sdk/communication/Azure.Communication.CallAutomation/src/Models/CustomContext.cs [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        /// <summary> Dictionary of VOIP headers. </summary>
+        public IDictionary<string, string> VoipHeaders { get; }
+
+        /// <summary> Dictionary of SIP headers. </summary>
+        public IDictionary<string, string> SipHeaders { get; }
+
+        /// <summary>
+        /// Creates a new CustomContext.
+        /// </summary>
+        public CustomContext(IDictionary<string, string> sipHeaders, IDictionary<string, string> voipHeaders)
```
Comment (minwoolee-msft) [inline sdk/communication/Azure.Communication.CallAutomation/src/Models/CustomContext.cs:23]: From the code, it seems that CustomerContext is instantiated by CallAutomation inside of Options, then customer would add new headers with ".Add".

But CustomContext also has public cosntructor - is there any instance where Developer would instantiate this class directly?

Comment (fangchen0601) [reply sdk/communication/Azure.Communication.CallAutomation/src/Models/CustomContext.cs:23]: change the constructor to 'internal', we create it for customer, customer do not need to initialize it manually.

**Review by minwoolee-msft — APPROVED** [review]

**Pull request merged**